src/main/java/com/exadel/discountwebapp/discount/service/DiscountService.java

```diff
@@ -61,4 +64,24 @@ public DiscountResponseVO update(Long id, DiscountRequestVO request) {
     public void deleteById(Long id) {
         discountRepository.deleteById(id);
     }
-}
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void addDiscountToFavorites(Long userId, Long discountId) {
+        Discount discount = discountRepository.findById(discountId)
+                .orElseThrow(() -> new EntityNotFoundException(Discount.class, "id", discountId));
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new EntityNotFoundException(User.class, "id", userId));
+        discount.getUserFavorites().add(user);
```

Comment:
What is the user already added this discount to favorites? Probably we should check it

Reply:
added check before adding discount to favorites